duckdns/data/run.sh

```diff
@@ -14,30 +14,57 @@ DOMAINS=$(bashio::config 'domains | join(",")')
 WAIT_TIME=$(bashio::config 'seconds')
 ALGO=$(bashio::config 'lets_encrypt.algo')
 
-# Function that performe a renew
+# Function that performs a renew
 function le_renew() {
     local domain_args=()
     local domains=''
-    local aliases=''
+    local aliases=()
+    local uniq_sorted_aliases=''
 
     domains=$(bashio::config 'domains')
 
     # Prepare domain for Let's Encrypt
     for domain in ${domains}; do
         for alias in $(jq --raw-output --exit-status "[.aliases[]|{(.alias):.domain}]|add.\"${domain}\" | select(. != null)" /data/options.json) ; do
-            aliases="${aliases} ${alias}"
+            aliases+=("${alias}")
```

Comment:
Can we make it work, if we will collect aliases separately for each domain?
So, we can move inside for-loop querying for aliases and sorting them:
```sh
function le_renew() {
    local domain_args=()
    local domains=''
    local aliases=''

    domains=$(bashio::config 'domains')

    # Prepare domain for Let's Encrypt
    for domain in ${domains}; do
        for alias in $(jq --raw-output --exit-status "[.aliases[]|select(.alias == \"${domain}\")|.domain+\" \"]|add | select(. != null)" /data/options.json) ; do
            aliases+=("${alias}")
        done
        uniq_sorted_aliases="$(echo "${aliases[@]}" | tr ' ' '\n' | sort | uniq | tr '\n' ' ')"
        bashio::log.info "Renew certificate for domain: ${domain} and aliases: $(echo -n "${uniq_sorted_aliases}")"
        cert="${CERT_DIR}/${domain}/cert.pem"
        if [ -f "${cert}" ]; then
            bashio::log.info "Checking existing cert..."
            certnames="$(openssl x509 -in "${cert}" -text -noout | grep DNS: | sed 's/DNS://g' | tr -d ' ' | tr ',' '\n' | sort | tr '\n' ' ' | sed 's/ $//')"
            givennames="$(echo "${domain}" "${uniq_sorted_aliases}"| tr ' ' '\n' | sort | tr '\n' ' ' | sed 's/ $//' | sed 's/^ //')"
            if [ "${certnames}" != "${givennames}" ]; then
                bashio::log.info "Certificate names do not match, requesting new certificate."
                issue_cert=1
            fi
            if ! (openssl x509 -checkend $((30 * 86400)) -noout -in "${cert}" > /dev/null 2>&1); then
                bashio::log.info "Certificate is due for renewal, auto-renewing."
                issue_cert=1
            fi
        else
            bashio::log.info "No certificate found for ${domain}, requesting new certificate."
            issue_cert=1
        fi

        if [ -n "${issue_cert:-}" ]; then
        # Order is important as we want the duckdns.org domain to be first
        for domain in $(echo "${domain}" "${uniq_sorted_aliases}" | tr ' ' '\n' | uniq | tr '\n' ' '); do
            domain_args+=("--domain" "${domain}")
            dehydrated --cron --algo "${ALGO}" --hook ./hooks.sh --challenge dns-01 "${domain_args[@]}" --out "${CERT_DIR}" --config "${WORK_DIR}/config" || true
        done
        else
            bashio::log.info "Certificate still valid. Skipping renew!"
        fi
    done
    
    LE_UPDATE="$(date +%s)"
}
```

I don't have enough knowledge yet to test everything, but looks like it should works when I look for separate pars.